test/Datadog.Trace.Tests/Logging/LoggingProviderTestHelpers.cs

```diff
@@ -18,91 +18,100 @@ internal static class LoggingProviderTestHelpers
         internal static readonly int CustomPropertyValue = 1;
         internal static readonly string LogPrefix = "[Datadog.Trace.Tests.Logging]";
 
-        private const string Log4NetExpectedStringFormat = "\"{0}\":\"{1}\"";
-        private const string SerilogExpectedStringFormat = "{0}: \"{1}\"";
-
         internal static Tracer InitializeTracer(bool enableLogsInjection)
         {
             var settings = new TracerSettings();
             var writerMock = new Mock<IAgentWriter>();
             var samplerMock = new Mock<ISampler>();
 
             settings.LogsInjectionEnabled = enableLogsInjection;
+            settings.Version = "custom-version";
+            settings.Environment = "custom-env";
 
             return new Tracer(settings, writerMock.Object, samplerMock.Object, scopeManager: null, statsd: null);
         }
 
-        internal static void PerformParentChildScopeSequence(Tracer tracer, ILog logger, Func<string, object, bool, IDisposable> openMappedContext, out Scope parentScope, out Scope childScope)
+        internal static void LogInSpanWithServiceName(Tracer tracer, ILog logger, Func<string, object, bool, IDisposable> openMappedContext, string service, out Scope scope)
         {
-            logger.Log(LogLevel.Info, () => $"{LogPrefix}Logged before starting/activating a scope");
-
-            parentScope = tracer.StartActive("parent");
-            logger.Log(LogLevel.Info, () => $"{LogPrefix}Started and activated parent scope.");
-
-            var customPropertyContext = openMappedContext(CustomPropertyName, CustomPropertyValue, false);
-            logger.Log(LogLevel.Info, () => $"{LogPrefix}Added custom property to MDC");
-
-            childScope = tracer.StartActive("child");
-            logger.Log(LogLevel.Info, () => $"{LogPrefix}Started and activated child scope.");
-
-            childScope.Close();
-            logger.Log(LogLevel.Info, () => $"{LogPrefix}Closed child scope and reactivated parent scope.");
-
-            customPropertyContext.Dispose();
-            logger.Log(LogLevel.Info, () => $"{LogPrefix}Removed custom property from MDC");
-
-            parentScope.Close();
-            logger.Log(LogLevel.Info, () => $"{LogPrefix}Closed child scope so there is no active scope.");
+            using (scope = tracer.StartActive("span", serviceName: service))
+            {
+                using (var mappedContext = openMappedContext(CustomPropertyName, CustomPropertyValue, false))
+                {
+                    logger.Log(LogLevel.Info, () => $"{LogPrefix}Entered single scope with a different service name.");
+                }
+            }
         }
 
```

Comment:
Is it worth adding a test here to check behavior of asynchronous child spans that outlive the parents?
I know there are some current issues with that, but it could give us a jumping point with the test if we don't fix it here.

Reply:
That would be good, but I'd like to postpone that to a follow-up PR. Is that okay with you?

Reply:
Leaving this for a future PR.